middleware/src/middleware.go

```diff
@@ -186,87 +184,71 @@ func (middleware *Middleware) DummyAdminPassword() string {
-
-	case rpcmessages.Mount:
-		log.Println("Executing a USB flashdrive mount via the cmd script")
-		out, err := middleware.runBBBCmdScript("flashdrive", "mount"+" "+args.Path)
-		if err != nil {
-			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
-		}
-		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
-
-	case rpcmessages.Unmount:
-		log.Println("Executing a USB flashdrive unmount via the cmd script")
-		out, err := middleware.runBBBCmdScript("flashdrive", "unmount")
-		if err != nil {
-			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
-		}
-		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+// MountFlashdrive returns an ErrorResponse struct in a response to a rpcserver request
+func (middleware *Middleware) MountFlashdrive() rpcmessages.ErrorResponse {
+	log.Println("Executing a USB flashdrive check via the cmd script")
+	outCheck, err := middleware.runBBBCmdScript("flashdrive", "check", "")
+	if err != nil {
+		return rpcmessages.ErrorResponse{Success: false, Message: string(outCheck), Code: err.Error()}
+	}
+	flashDriveName := strings.TrimSuffix(string(outCheck), "\n")
 
-	default:
-		errorMessage := fmt.Sprintf("Method %d not supported for Flashdrive().", args.Method)
-		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
+	log.Println("Executing a USB flashdrive mount via the cmd script")
+	outMount, err := middleware.runBBBCmdScript("flashdrive", "mount", flashDriveName)
+	if err != nil {
+		return rpcmessages.ErrorResponse{Success: false, Message: string(outMount), Code: err.Error()}
 	}
+	return rpcmessages.ErrorResponse{Success: true}
 }
 
-// Backup returns a GenericResponse struct in response to a rpcserver request
-func (middleware *Middleware) Backup(method rpcmessages.BackupArgs) (rpcmessages.GenericResponse, error) {
-	switch method {
-	case rpcmessages.BackupSysConfig:
-		log.Println("Executing a backup of the system config via the cmd script")
-		out, err := middleware.runBBBCmdScript("backup", "sysconfig")
-		if err != nil {
-			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
-		}
-		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
-
-	case rpcmessages.BackupHSMSecret:
-		log.Println("Executing a backup of the c-lightning hsm_secret via the cmd script")
-		out, err := middleware.runBBBCmdScript("backup", "hsm_secret")
-		if err != nil {
-			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
-		}
-		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+// UnmountFlashdrive returns an ErrorResponse struct in a response to a rpcserver request
+func (middleware *Middleware) UnmountFlashdrive() rpcmessages.ErrorResponse {
+	log.Println("Executing a USB flashdrive unmount via the cmd script")
+	out, err := middleware.runBBBCmdScript("flashdrive", "unmount", "")
```

Comment:
`UnmountFlashdrive` doesn't do anything like `middleware.runBBBCmdScript("flashdrive", "check", "")` like `MountFlashdrive` does. What happens if there are multiple drives plugged in, does it unmount all of them?

Reply:
Good point, but only the drive mounted at `/mnt/backup` is unmounted by the script. 

https://github.com/digitalbitbox/bitbox-base/blob/master/armbian/base/scripts/bbb-cmd.sh#L231-L239